src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdStateUpdateService.java

```diff
@@ -0,0 +1,94 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.State;
+import uk.gov.hmcts.reform.sscs.model.single.hearing.HearingGetResponse;
+
+import static java.util.Objects.nonNull;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.AWAITING_LISTING;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingCaseStatus.LISTED;
+import static uk.gov.hmcts.reform.sscs.model.single.hearing.ListingStatus.FIXED;
+
+@Service
+public class CcdStateUpdateService {
+
+    public void updateListed(HearingGetResponse hearingResponse, SscsCaseData sscsCaseData) {
+        State state = State.UNKNOWN;
+        if (isHearingListingStatusFixed(hearingResponse)) {
+            state = mapHmcCreatedOrUpdatedToCcd(hearingResponse);
+        }
+
+        if (!state.equals(State.UNKNOWN)) {
+            sscsCaseData.setState(state);
+        }
+    }
+
+    public void updateCancelled(HearingGetResponse hearingResponse, SscsCaseData sscsCaseData) {
+        State state = State.UNKNOWN;
+
+        if (isHearingCancelled(hearingResponse)) {
+            state = mapHmcCancelledToCcdState(
+                hearingResponse.getHearingResponse().getHearingCancellationReason());
+        }
+        if (!state.equals(State.UNKNOWN)) {
+            sscsCaseData.setState(state);
+        }
+    }
+
+    public void updateFailed(SscsCaseData sscsCaseData) {
+        sscsCaseData.setState(State.HANDLING_ERROR);
+    }
+
+    private State mapHmcCreatedOrUpdatedToCcd(HearingGetResponse hearingResponse) {
+        if (isHearingListed(hearingResponse)) {
+            return State.HEARING;
```

Comment:
Can we confirm if this is the right state?

Reply:
It is in the requirements:

1. Booked hearing:

Critical attributes:

hearingResponse.listingStatus = "Fixed"

THEN

hearingResponse.laCaseStatus = "Listed"

hearingDaySchedule.hearingStartDateTime 
hearingDaySchedule.hearingEndDateTime
hearingDaySchedule.hearingVenueId
**CCD State = Hearing**